src/pal/src/exception/signal.cpp

```diff
@@ -240,6 +240,52 @@ void SEHCleanupSignals()
 /* internal function definitions **********************************************/
 
 #if !HAVE_MACH_EXCEPTIONS
+
+static void invoke_previous_action(struct sigaction* action, int code, siginfo_t *siginfo, void *context, bool signalRestarts = true)
+{
+    _ASSERTE(action != NULL);
+
+    if (action->sa_handler == SIG_IGN)
```

Comment:
The sigaction man page says:
>       On some architectures a union is involved: do not assign to both
>       sa_handler and sa_sigaction.

So it seems we should check the sa_flags to see if the sa_handler or sa_sigaction is to be used.

Reply:
On Linux for example they map to the same field which means setting one to NULL clears out the other, which is a problem.
I think when getting there is no need to check sa_flags. They will both return the same value.

Reply:
I'll add it. Better be safe than sorry :)